src/types/emr/medicationStatement.ts

```diff
@@ -0,0 +1,49 @@
+import { Period } from "@/types/questionnaire/base";
+import { Code } from "@/types/questionnaire/code";
+
+export const MEDICATION_STATEMENT_INFORMATION_SOURCE_TYPE = [
+  "patient",
+  "user",
+  "related_person",
+] as const;
+
+export type MedicationStatementInformationSourceType =
+  (typeof MEDICATION_STATEMENT_INFORMATION_SOURCE_TYPE)[number];
+
+export type MedicationStatementInformationSource = {
+  type: MedicationStatementInformationSourceType;
+  id?: string; // UUID
+  display?: string;
+  relationship?: string;
+};
+
+export const MEDICATION_STATEMENT_STATUS = [
+  "active",
+  "on_hold",
+  "completed",
+  "stopped",
+  "unknown",
+  "entered_in_error",
+  "not_taken",
+  "intended",
+] as const;
+
+export type MedicationStatementStatus =
+  (typeof MEDICATION_STATEMENT_STATUS)[number];
+
+export type MedicationStatement = {
+  readonly id: string;
+  status: MedicationStatementStatus;
+  reason?: string;
+
+  medication: Code;
+  dosage?: string;
```

Comment:
Call it `dosage_text` so that we are clear that we storing the information as text